src/v2/Components/UserSettings/TwoFactorAuthentication/__tests__/TwoFactorAuthentication.jest.tsx

```diff
@@ -197,3 +199,47 @@ describe("TwoFactorAuthentication", () => {
     })
   })
 })
+
+describe("Two factor authentication enrollment", () => {
+  const props = {
+    backupSecondFactors: ['d3bd78d468', '7aa4c5922c'],
+    factorTypeName: 'AppSecondFactor'
+  }
+  const getWrapper = (passedProps = props) => {
+    return mount(<BackupSecondFactorReminder {...passedProps} />)
+  }
+
+  const getCopyButton = () => {
+    return getWrapper().find('[data-test="copyButton"]').first()
+  }
+
+  describe("when the browser does not support clipboard", () => {
+    it("displays a copy button based on browser support", () => {
+      expect(getCopyButton()).toHaveLength(0)
+    })
+  })
+
+  describe("when the browser supports clipboard", () => {
+    const mockClipboard = { writeText: jest.fn() };
+    beforeEach(() => {
+      Object.assign(navigator, {
+        clipboard: mockClipboard
+      });
+    })
+
+    it("displays a copy button", () => {
+      const copyButton = getCopyButton()
+
+      expect(copyButton).toHaveLength(1)
+      expect(copyButton.text()).toBe("Copy")
+    })
+
+    it("enables user to copy the recovery codes", () => {
+      const copyButtonProps = getCopyButton().props()
+
+      if (copyButtonProps.onClick) copyButtonProps.onClick({} as any);
+
+      expect(navigator.clipboard.writeText).toBeCalledTimes(1)
+    })
+  })
+})
```

Comment:
@mc-jones I added these tests to cover the newly added behavior and I have it on `TwoFactorAuthentication` tests at this point as we discussed yesterday. However, these are not being benefited from the setup that is required for the other tests that are present in this same file.

At this point, they feel much more like unit tests for me, and I would be inclined to actually have an exclusive file for these on `src/v2/Components/UserSettings/TwoFactorAuthentication/Components/BackupSecondFactorReminder/__tests__/index.jest.tsx` (following similar pattern that exists today for `BackupSecondFactor` component [here](https://github.com/artsy/force/blob/master/src/v2/Components/UserSettings/TwoFactorAuthentication/Components/BackupSecondFactor/__tests__/index.jest.tsx))

What do you think?

Reply:
@lidimayra - Yeah, I definitely think that is correct. 👍

Reply:
Thanks for the response, @mc-jones 🙌 

I ended up extracting this whole behavior (and consequently the tests as well) to a new component (1b1034a) so we can reuse it on both modals.